### PR TITLE
Support of Polkadot Addresses

### PR DESCRIPTION
**Pull request by borispovod**
* Supports ss58 address.
* Choose different dialects using dove (using manifest or -d/--dialect), e.g.:
`dove new -d polkadot tst`
* Manifest deserialize/serialize address as string now, later i use dialect to get real account address
* Default dialect - dfinance.
* **Polkadot dialect doesn't work because of address length of 34 bytes.**

### CODE REVIEW

**Review by RIg410 — APPROVED** [review]

File: dove/src/context.rs [hunk 1]
```diff
@@ -35,18 +35,27 @@ impl Context {
 
     /// Returns provided account address.
     pub fn account_address(&self) -> Result<ProvidedAccountAddress> {
-        self.dialect
-            .normalize_account_address(&format!("0x{}", &self.manifest.package.account_address))
+        let acc_addr = self.manifest.package.account_address.clone().unwrap();
```
Comment (fzzr-) [inline dove/src/context.rs:38]: Will be great to remove that `unwrap` and return `Err` if cannot `clone`.

File: dove/src/index/resolver/git.rs [hunk 2]
```diff
@@ -70,7 +72,18 @@ fn get_dep_address(path: &Path) -> Result<Option<AccountAddress>, Error> {
...
+
+        let dialect_name = manifest
+            .package
+            .dialect
+            .clone()
+            .unwrap_or_else(default_dialect);
+        let dialect = DialectName::from_str(&dialect_name)?.get_dialect();
+
+        let provided_account_address =
+            dialect.normalize_account_address(&manifest.package.account_address.unwrap())?;
```
Comment (fzzr-) [inline dove/src/index/resolver/git.rs:84]: Can we remove that `unwrap` and return `Err` if there's no `account_address`?

File: lang/src/compiler/ss58.rs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    addr[..2].copy_from_slice(&[bs58[0], 0]);
+    addr[2..].copy_from_slice(&bs58[1..PUB_KEY_LENGTH / 2 + 1]);
+    Ok(format!("0x{}", hex::encode_upper(addr)))
+}
+
+pub fn replace_ss58_addresses(source: &str, file_source_map: &mut FileOffsetMap) -> String {
+    let mut transformed_source = source.to_string();
+
+    for mat in SS58_REGEX.captures_iter(source).into_iter() {
+        let item = mat.get(0).unwrap();
```
Comment (fzzr-) [inline lang/src/compiler/ss58.rs:45]: I know we unable return `Result` here now, but I do not recommend to use just `unwrap` without explicit explanation what is happen here. So, IMHO, will be better change `unwrap` to `expect("your error message")`.

**Review by fzzr- — CHANGES_REQUESTED** [review]

**Comment by borispovod** [comment]
don't merge and check it yet, want to write few tests, almost done

**Comment by borispovod** [comment]
now it's ready for review :)

**Review by fzzr- — APPROVED** [review]
LGTM 👍
Thanks for the tests!

**Pull request merged**